recipes/eleuther_eval.py

```diff
@@ -89,17 +96,63 @@ def tok_encode(self, text: str, **kwargs) -> List[int]:
         # https://github.com/pytorch-labs/gpt-fast/blob/main/eval.py#L123.
         return self._tokenizer.encode(text=text, add_bos=False, add_eos=False)
 
-    def tok_decode(self, tokens: List[int], **kwargs) -> str:
+    def tok_batch_encode(
+        self, text: List[str], **kwargs
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        tokenized_text = [self.tok_encode(x) for x in text]
+
+        # pad left
+        x = pad_sequence(
+            [
+                torch.tensor(x[::-1]) for x in tokenized_text
+            ],  # first flip each sequence and pad
+            batch_first=True,
+            padding_value=self._tokenizer.pad_id,
+        ).flip(
+            dims=[1]
+        )  # flip back to correct order
+
+        return x, torch.ones_like(x)  # return 'mask' b/c it's expected by the harness
+
+    def tok_decode(self, tokens: Union[List[int], int], **kwargs) -> str:
+        if isinstance(tokens, int):
+            tokens = [tokens]
         return self._tokenizer.decode(tokens)
 
     def _model_call(self, inps: torch.Tensor, **kwargs) -> torch.Tensor:
         return self._model(inps)
 
-    def _model_generate(self, *args, **kwargs):
-        raise RuntimeError(
-            "This recipe does not currently support tasks that evaluate free generation,"
-            "e.g. `truthfulqa_gen` or `bigbench_color_generate_until`."
+    def _model_generate(
+        self, context: torch.Tensor, **generation_kwargs
+    ) -> torch.Tensor:
+        curr_batch_size = context.size(0)
+
+        # Setup caches for a given batch size
+        # Technically this is not necessary, but it's a good way to ensure that
+        # the caches won't error on a different batch size. In addition, caches
+        # are not needed for a regular model call, so we just setup here
+        with context.device:
+            self._model.setup_caches(batch_size=curr_batch_size, dtype=self._dtype)
+
+        temperature = generation_kwargs.get("temperature", 0.0)
```

Comment:
Is this a reasonable default for temperature?

Reply:
This is the MOST common default for temperature b/c it greedily takes the best token.